boltdb.go

```diff
@@ -62,60 +62,66 @@ func NewBoltDBWithOpts(name string, dir string, opts *bbolt.Options) (DB, error)
 	return &BoltDB{db: db}, nil
 }
 
-func (bdb *BoltDB) Get(key []byte) (value []byte) {
+func (bdb *BoltDB) Get(key []byte) (value []byte, err error) {
 	key = nonEmptyKey(nonNilBytes(key))
-	err := bdb.db.View(func(tx *bbolt.Tx) error {
+	err = bdb.db.View(func(tx *bbolt.Tx) error {
 		b := tx.Bucket(bucket)
 		if v := b.Get(key); v != nil {
 			value = append([]byte{}, v...)
 		}
 		return nil
 	})
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 	return
 }
 
 func (bdb *BoltDB) Has(key []byte) bool {
-	return bdb.Get(key) != nil
+	bytes, err := bdb.Get(key)
+	if err != nil {
```

Comment:
This may be problematic cause it hides errors. In other words, this function should not IMHO just return false if an error is raised, it really should panic

Reply:
I was thinking about this as well, could return the error and let the application handle it, I want to avoid panicking in this case. I can switch out the `bool` for an `error` didn't know how this would be accepted as the name `Has()` to me sounds like it should return true or false, but if people are up for it returning the error I will make the change

Reply:
I think in keeping with the rest of the functions having the signature be:
```go
func (bdb *BoltDB) Has(key []byte) (b bool, err error) {}
```

Reply:
IMHO @jackzampolin suggestion is the way to go

Reply:
sounds good